crates/optimism/payload/src/builder.rs

```diff
@@ -542,16 +547,40 @@ impl ExecutionInfo {
             executed_senders: Vec::with_capacity(capacity),
             receipts: Vec::with_capacity(capacity),
             cumulative_gas_used: 0,
+            cumulative_da_bytes_used: 0,
             total_fees: U256::ZERO,
         }
     }
+
+    /// Returns true if the transaction would exceed the block limits.
+    pub fn is_tx_over_limits(
+        &self,
+        tx: &OpTransactionSigned,
+        block_gas_limit: u64,
+        tx_data_limit: Option<u64>,
+        block_data_limit: Option<u64>,
+    ) -> bool {
+        if tx_data_limit.is_some_and(|da_limit| tx.length() as u64 > da_limit) {
+            return true;
+        }
+
+        if block_data_limit
+            .is_some_and(|da_limit| self.cumulative_da_bytes_used + (tx.length() as u64) > da_limit)
+        {
+            return true;
+        }
```

Comment:
this now uses the length instead of the flz compressed length

should we change this here? or is this fine for this pr?

Reply:
ah sorry, I may have misunderstood. I think it's OK to use uncompressed size temporarily until we implement DA cost tracking. I thought the comment above meant to remove the compressed size calculation for now, but I could add it back in without modifying the pooled tx type.

Reply:
I wrote an example implementation of this here, but I'm a little concerned about changing revm versions to a Git hash: dbe1610053dda5254cf917b3bb27a9515fb6c51b

If this looks OK, I can add it to this PR, but if this isn't urgent, we can just implement after the issue you created.

Reply:
meh that's unfortunate that this wasn't included in the latest revm release...

https://github.com/bluealloy/revm/pull/1985